src/features/TransactionStatus/TransactionStatus.tsx

```diff
@@ -0,0 +1,137 @@
+  makePendingTxReceipt,
+  makeTxConfigFromTxReceipt
+} from '@utils/transaction';
+import { noOp } from '@utils';
+import { useEffectOnce, useUpdateEffect } from '@vendor';
+import { DEFAULT_NETWORK, ROUTE_PATHS } from '@config';
+import { translateRaw } from '@translations';
+import { getTxsFromAccount } from '@services/Store/helpers';
+
+const SUPPORTED_NETWORKS: NetworkId[] = ['Ethereum', 'Ropsten', 'Goerli', 'Kovan', 'ETC'];
+
+const TransactionStatus = withRouter(({ history, match, location }) => {
+  const qs = queryString.parse(location.search);
+
+  const { assets } = useContext(AssetContext);
+  const { getNetworkById, networks } = useContext(NetworkContext);
+  const { accounts } = useContext(StoreContext);
+
+  const defaultTxHash = match.params.txHash ? match.params.txHash : '';
+  const defaultNetwork =
+    qs.network && SUPPORTED_NETWORKS.includes(qs.network) ? qs.network : DEFAULT_NETWORK;
+
+  const [txHash, setTxHash] = useState(defaultTxHash);
+  const [networkId, setNetwork] = useState<NetworkId>(defaultNetwork);
+  const [tx, setTx] = useState<{ config: ITxConfig; receipt: ITxReceipt } | undefined>(undefined);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState('');
+
+  const network = networkId && getNetworkById(networkId);
+
+  const txCache = getTxsFromAccount(accounts);
+  const cachedTx = txCache.find(
+    (t) => t.hash === (txHash as ITxHash) && t.asset.networkId === networkId
+  );
+
+  // Fetch TX on load if possible
+  useEffectOnce(() => {
+    if (!isEmpty(defaultTxHash)) {
+      fetchTx();
+    }
+  });
+
+  // Update URL
+  useUpdateEffect(() => {
+    if (networkId === DEFAULT_NETWORK) {
+      history.replace(`${ROUTE_PATHS.TX_STATUS.path}/${txHash}`);
+    } else {
+      history.replace(`${ROUTE_PATHS.TX_STATUS.path}/${txHash}?network=${networkId}`);
+    }
+  }, [txHash, networkId]);
+
+  const fetchTx = async () => {
+    setLoading(true);
+    try {
+      const txResult = await (async () => {
+        if (!cachedTx) {
+          const provider = new ProviderHandler(network);
+          const fetchedTx = await provider.getTransactionByHash(txHash as ITxHash, true);
+          if (!fetchedTx) {
+            return undefined;
+          }
+          const fetchedTxConfig = makeTxConfigFromTransactionResponse(
+            fetchedTx,
+            assets,
+            network,
+            accounts
+          );
+          return {
+            config: fetchedTxConfig,
+            receipt: makePendingTxReceipt(txHash)(ITxType.UNKNOWN, fetchedTxConfig)
+          };
+        } else {
+          return {
+            config: makeTxConfigFromTxReceipt(cachedTx, assets, networks, accounts),
+            receipt: cachedTx
+          };
+        }
+      })();
+      if (!txResult) {
+        setError(translateRaw('TX_NOT_FOUND'));
+      } else {
+        setTx(txResult);
+      }
+    } catch (err) {
+      console.error(err);
+    }
+    setLoading(false);
```

Comment:
Use `finally` to ensure it's executed at the expected moment.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch